### PR TITLE
change action mailer to wrap content correctly when attachments and inline exist

### PR DESCRIPTION
**Pull request by HookyQR**

### CODE REVIEW

File: actionmailer/lib/action_mailer/base.rb [hunk 1]
```diff
@@ -932,6 +934,23 @@ def create_parts_from_responses(m, responses) #:nodoc:
       end
     end
 
+    def wrap_inline_attachments(message)
+      # If we have both types of attachment, wrap all the inline attachments
+      # in multipart/related, but not the actual attachments
+      return message unless message.attachments.detect(&:inline?) && message.attachments.detect { |a| !a.inline? }
```
Comment (dylanmckay) [inline actionmailer/lib/action_mailer/base.rb]: Place this predicate into a method, and also use it above in `Base#set_content_type`

File: actionmailer/lib/action_mailer/base.rb [hunk 2]
```diff
@@ -932,6 +934,23 @@ def create_parts_from_responses(m, responses) #:nodoc:
...
+      m = Mail.new
+      m.header = message.header.to_s
+      # copy bcc manually because it is omitted in header.to_s
+      m.bcc = message.header[:bcc].value if message.header[:bcc]
+      m.content_type = nil
```
Comment (dylanmckay) [inline actionmailer/lib/action_mailer/base.rb]: Shouldn't we be getting the `Content-Type` from the given `message`?

Comment (HookyQR) [inline actionmailer/lib/action_mailer/base.rb]: The content type gets set in `set_content_type` which gets called after this method no matter what.

File: actionmailer/lib/action_mailer/base.rb (hunk 1 above)
Comment (HookyQR) [inline actionmailer/lib/action_mailer/base.rb]: This is (almost completely) a copy of the PR against the rails/rails repo. I've deliberately left it structured the same.

**Pull request merged**